communication/examples/local-echo.rs

```diff
@@ -0,0 +1,242 @@
+                                        println!("Echoed message");
+                                    } else {
+                                        println!("Error sending echo: {:?}", response.unwrap_err());
+                                    }
+                                }
+                            }
+                        }
+                        P2PReqResEvent::Res {
+                            peer_id,
+                            request_id,
+                            response,
+                        } => println!(
+                            "Response from peer {:?} for Request {:?}:\n{:?}",
+                            peer_id, request_id, response
+                        ),
+                        error => {
+                            println!("Error {:?}", error)
+                        }
+                    },
+                    P2PEvent::Identify(event) => {
+                        if let P2PIdentifyEvent::Received {
+                            peer_id,
+                            info: _,
+                            observed_addr,
+                        } = event.deref().clone()
+                        {
+                            println!(
+                                "Received identify event: {:?} observes us at {:?}",
+                                peer_id, observed_addr
+                            );
+                        }
+                    }
+                    _ => {}
+                },
+                Poll::Ready(None) => {
+                    return Poll::Ready(());
+                }
+                Poll::Pending => {
+                    if !listening {
+                        let mut listeners = Swarm::listeners(&swarm).peekable();
+                        if listeners.peek() == None {
+                            println!("No listeners. The port may already be occupied.")
+                        } else {
+                            println!("Listening on:");
+                            for a in listeners {
+                                println!("{:?}", a);
+                            }
+                        }
+                        println!("commands:");
+                        println!("PING <peer_id>");
+                        println!("MSG <peer_id>");
+                        println!("DIAL <peer_addr>");
+                        println!("LIST");
+                        if cfg!(not(feature = "mdns")) {
+                            println!("Since mdns is not enabled, peers have to be DIALed first to connect before PING / MSG them");
+                        }
+                        listening = true;
+                    }
+                    break;
+                }
+            }
+        }
+        Poll::Pending
+    }));
+    Ok(())
+}
+
+fn handle_input_line(swarm: &mut Swarm<P2PNetworkBehaviour<Request, Response>>, line: String) {
+    if let Some(peer_id) = Regex::new("PING\\s+\"(\\w+)\"")
```

Comment:
Little verbose on the if branches.  Would be better if you put most of the internal logic into the body of the if checks so that it doesn't have to execute it every time.